runtime/drivers/duckdb/connectors.go

```diff
@@ -201,3 +150,111 @@ func resolveLocalPath(env *connectors.Env, path, sourceName string) (string, err
 	}
 	return finalPath, nil
 }
+
+func sourceReader(paths []string, properties map[string]interface{}) (string, error) {
+	format, formatDefined := properties["format"].(string)
+	if formatDefined {
+		format = fmt.Sprintf(".%s", format)
+	} else {
+		format = fileutil.FullExt(paths[0])
+	}
+	// Generate a "read" statement
+	if containsAny(format, []string{".csv", ".tsv", ".txt"}) {
+		// CSV reader
+		return generateReadCsvStatement(paths, properties)
+	} else if strings.Contains(format, ".parquet") {
+		// Parquet reader
+		return generateReadParquetStatement(paths, properties)
+	} else if containsAny(format, []string{".json", ".ndjson"}) {
+		// JSON reader
+		return generateReadJSONStatement(paths, properties)
+	} else {
+		return "", fmt.Errorf("file type not supported : %s", format)
+	}
+}
+
+func containsAny(s string, targets []string) bool {
+	source := strings.ToLower(s)
+	for _, target := range targets {
+		if strings.Contains(source, target) {
+			return true
+		}
+	}
+	return false
+}
+
+func generateReadCsvStatement(paths []string, properties map[string]interface{}) (string, error) {
+	ingestionProps := collectDuckDBIngestionProperties("duckdb.csv.", properties)
+	// backward compatibility: csv.delimiter might be passed separately from duckdb.csv.delim and has a priority
+	if csvDelimiter, csvDelimiterDefined := properties["csv.delimiter"]; csvDelimiterDefined {
+		ingestionProps["delim"] = fmt.Sprintf("'%v'", csvDelimiter)
+	}
```

Comment:
I would put these backwards compatibility rules in `runtime/services/catalog/artifacts/yaml/objects.go` instead. So right at artifact parse time, `csv.delimiter` gets mapped to `duckdb.delim`.

Reply:
moved to `objects.go`